examples/text/plot_document_clustering.py

```diff
@@ -6,148 +6,32 @@
 This is an example showing how the scikit-learn can be used to cluster
 documents by topics using a bag-of-words approach. This example uses
 a scipy.sparse matrix to store the features instead of standard numpy arrays.
-
-Two feature extraction methods can be used in this example:
-
-  - TfidfVectorizer uses a in-memory vocabulary (a python dict) to map the most
-    frequent words to features indices and hence compute a word occurrence
-    frequency (sparse) matrix. The word frequencies are then reweighted using
-    the Inverse Document Frequency (IDF) vector collected feature-wise over
-    the corpus.
-
-  - HashingVectorizer hashes word occurrences to a fixed dimensional space,
-    possibly with collisions. The word count vectors are then normalized to
-    each have l2-norm equal to one (projected to the euclidean unit-ball) which
-    seems to be important for k-means to work in high dimensional space.
-
-    HashingVectorizer does not provide IDF weighting as this is a stateless
-    model (the fit method does nothing). When IDF weighting is needed it can
-    be added by pipelining its output to a TfidfTransformer instance.
-
-Two algorithms are demoed: ordinary k-means and its more scalable cousin
```

Comment:
A note here: I checked locally and for this example, there is no point in using `MiniBatchKMeans`.
At the end of the example, we can add a remark that, with a large number of data samples, it would be best to use the `MiniBatchKMeans` to improve the computational performance.